### DISCUSSION TITLE
[Enhancement] Adding the ability to make Splash Screens cross platform solution

### DISCUSSION THREAD

**Discussion by AhmedAbuelnour** [dotnet/maui#308] [Ideas]
## Summary
Adding a new feature to enable to make cross platform Splash screen,

**Comment by Daniellled** [comment]
Based on the MAUI preview 4 release notes it looks like the splash screen, MauiSplashScreen, is only supported on specific platforms and is static. - https://devblogs.microsoft.com/dotnet/announcing-net-maui-preview-4/#splash-screen

1. Will Splash Screens be supported on Windows platforms? (or is it and I just missed it)
2. It looks like the MauiSplashScreen project setting only allows for a static splash screen. Will a more dynamic/rich splash screen be added in the future? To perhaps display information to the user about the start up sequence of the application?

**Reply by GiannKarlo** [reply]
Hi , is there any update on this ? I still could not see my splashscreen on windows. :(

**Comment by vhugogarcia** [comment]
I think it will be pretty limited from what I saw. Maybe if they allow gradients as background that will make tons of differences. I wish I could contribute on this. Sadly, I dont have experience with that.

**Comment by MartyIX** [comment]
I would love to see splash screens on Windows too 👍 

edit: Is there a workaround to show splash screen on Windows at the moment?

**Reply by GiannKarlo** [reply]
Any update on this ?

**Comment by GiannKarlo** [comment]
up

**Comment by ali-h2010** [comment]
### My Workaround
I used a MediaElement to show the animated splash screen as a video. Set the initial page to this Content page

```
<?xml version="1.0" encoding="UTF-8" ?>
<ContentPage
    x:Class="AppSample.View.CustomSplashScreen.AnimatedSplashScreen"
    xmlns="http://xamarin.com/schemas/2014/forms"
    xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
    xmlns:xct="clr-namespace:Xamarin.CommunityToolkit.UI.Views;assembly=Xamarin.CommunityToolkit">
    <ContentPage.Content>
        <StackLayout HorizontalOptions="FillAndExpand" VerticalOptions="FillAndExpand">
            <xct:MediaElement ShowsPlaybackControls="False"
                              Source="ms-appx:///splash_screen_animated.mp4"
                              Aspect="AspectFill"
                              AutoPlay="True"
                              BackgroundColor="White"
                              />
        </StackLayout>
    </ContentPage.Content>
</ContentPage>

```

And on code behind, i am dismissing the splash screen and setting the main page after awaiting for the legth of the video

  

>       protected async override void OnAppearing()
>         {
>             base.OnAppearing();
>             await Task.Delay(TimeSpan.FromSeconds(5));
> 
>             App.SetMainPage();
>         }
